Update README with new python and botocore requirements (#369)
ansible-collections/amazon.aws: README.md

## Python version compatibility

> As the AWS SDK for Python (Boto3 and Botocore). has [ceased supporting Python 2.7](https://aws.amazon.com/blogs/developer/announcing-end-of-support-for-python-2-7-in-aws-sdk-for-python-and-aws-cli-v1/), this collection requires Python 3.6 or greater.

Reviewer: ```suggestion
As the AWS SDK for Python (Boto3 and Botocore) has [ceased supporting Python 2.7](https://aws.amazon.com/blogs/developer/announcing-end-of-support-for-python-2-7-in-aws-sdk-for-python-and-aws-cli-v1/), this collection requires Python 3.6 or greater.
```

> Starting with the 2.0.0 releases of both collections, it is generally the collection's policy to support the versions of `botocore` and `boto3` that were released 12 months prior to the most recent major collection release, following semantic versioning (for example, 2.0.0, 3.0.0).

Reviewer 2: Is it always clear to readers what 'both' collections are? If not:
```suggestion
Starting with the 2.0.0 releases of amazon.aws and community.aws, it is generally the collection's policy to support the versions of `botocore` and `boto3` that were released 12 months prior to the most recent major collection release, following semantic versioning (for example, 2.0.0, 3.0.0).
```

Reviewer 3: I guess so.

Reviewer 3: Mentioning it explicitly won't hurt.

## Python version compatibility

This collection depends on the AWS SDK for Python (Boto3 and Botocore). Starting with the 2.0 releases of both collections, it is generally the collection's policy to support the versions of `botocore` and `boto3` that were released 12 months prior to the most recent major collection release, following semantic versioning (for example, 2.0.0, 3.0.0).

> As AWS has [ceased supporting Python 2.7](https://aws.amazon.com/blogs/developer/announcing-end-of-support-for-python-2-7-in-aws-sdk-for-python-and-aws-cli-v1/), this collection requires Python 3.6 or greater.

Reviewer: ```suggestion
As the AWS SDK for Python (Boto3 and Botocore). has [ceased supporting Python 2.7](https://aws.amazon.com/blogs/developer/announcing-end-of-support-for-python-2-7-in-aws-sdk-for-python-and-aws-cli-v1/), this collection requires Python 3.6 or greater.

Starting with the 2.0.0 releases of both collections, it is generally the collection's policy to support the versions of `botocore` and `boto3` that were released 12 months prior to the most recent major collection release, following semantic versioning (for example, 2.0.0, 3.0.0).

Version 2.0.0 of this collection supports `boto3 >= 1.13.0` and `botocore >= 1.16.0`

Reviewer: It feels a little odd specifying the modules we need at the start of the "Python version" section before defining which Python version we require.
